### PR TITLE
Signer module

### PR DESCRIPTION
**Pull request by siburu**

### CODE REVIEW

File: pkg/relay/ethereum/signers/hd/config.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	fmt "fmt"
+	"math/big"
+
+	"github.com/datachainlab/ethereum-ibc-relay-chain/pkg/relay/ethereum"
+)
+
+var _ ethereum.SignerConfig = (*SignerConfig)(nil)
+
+func (c *SignerConfig) Validate() error {
```
Comment (bluele) [inline pkg/relay/ethereum/signers/hd/config.go:12]: Can we further validate `Mnemonic` and `Path` format using `GetPrvKeyFromMnemonicAndHDWPath`?

File: pkg/relay/ethereum/config.go [hunk 2]
```diff
@@ -42,10 +40,20 @@ func (c ChainConfig) Validate() error {
 	if c.MaxRetryForInclusion == 0 {
 		errs = append(errs, fmt.Errorf("config attribute \"max_retry_for_inclusion\" is zero"))
 	}
+	if err := c.Signer.GetCachedValue().(SignerConfig).Validate(); err != nil {
```
Comment (bluele) [inline pkg/relay/ethereum/config.go:43]: First you should check if `c.Signer` is non-nil.

**Review by bluele — COMMENTED** [review]
@siburu I left some comments about the config validation.

File: pkg/relay/ethereum/signers/hd/config.go (hunk 1 above)
Comment (siburu) [reply pkg/relay/ethereum/signers/hd/config.go:13]: @bluele Thank you! I fixed it now.

File: pkg/relay/ethereum/config.go (hunk 2 above)
Comment (siburu) [reply pkg/relay/ethereum/config.go]: @bluele Thank you! I fixed it now.

**Review by bluele — APPROVED** [review]
@siburu LGTM👍

**Pull request merged**